src/system/Form/RadioBoxGroup.js

```diff
@@ -3,84 +3,148 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
-import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
-import { MdCheckCircle } from 'react-icons/md';
+import ScreenReaderText from '../ScreenReaderText';
 
 /**
  * Internal dependencies
  */
-import { Heading, Text } from '../';
 
-const RadioBoxGroup = React.forwardRef(
-	( { onChange, groupLabel, value, options, ...props }, forwardRef ) => (
-		<RadioGroupPrimitive.Root
-			onValueChange={ onChange }
-			value={ value }
-			aria-label={ groupLabel }
-			sx={ {
-				display: 'flex',
-				gap: 2,
-			} }
-			ref={ forwardRef }
-			{ ...props }
+const RadioOption = ( {
+	disabled,
+	defaultValue,
+	option: { id, value, className, label, description, labelProps = {}, ...restOption },
```

Comment:
can you change id to `forLabel`? This is what we are using for all inputs and controls.